inference-engine/src/multi_device/multi_device.cpp

```diff
@@ -479,12 +479,17 @@ ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(co
     std::unordered_map<std::string, InferenceEngine::Parameter> multiNetworkConfig;
     multiNetworkConfig.insert(*priorities);
 
+    auto function = network.getFunction();
+    if (function == nullptr) {
+        THROW_IE_EXCEPTION << "MULTI device supports just ngraph network representation";
+    }
+
     DeviceMap<ExecutableNetwork> executableNetworkPerDevice;
     for (auto& p : metaDevices) {
         auto & deviceName = p.deviceName;
         auto & deviceConfig = p.config;
-        auto clonedNetwork = cloneNetwork(network);
-        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(CNNNetwork{clonedNetwork}, deviceName, deviceConfig) });
+        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(
+            CNNNetwork{ngraph::clone_function(*function)}, deviceName, deviceConfig) });
```

Comment:
do we need to clone network? I suppose we can rely on the fact that plugins MUST not change the function

@myshevts how do you think?

Reply:
I agree that we shouldn't clone function because plugins shouldn't change the original function. I propose to pass the original function to this map.

Reply:
@ilyachur notice that in 2021.3 for every plugin used in the specific setup, the MULTI will actually do the LoadNetwork in parallel.
Do we have general tests that:
1. Loading the same ngraph function multiple times doens't change that
2. Loading the same ngraph function from multiple threads doens't change that

Reply:
@myshevts Here you can find threading tests: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/shared/include/behavior/core_threading_tests.hpp

LoadNetwork works with constant networks, it means that plugins should apply all changes on cloned networks.

Reply:
I see, but in the tests you are referring too, the the ngraph func is loaded in parallel to the **same** device. So each load instance is making some changes, but these are the same, can this may went uncaught?

Reply:
I suppose multithreading and the fact that plugin must not change original network - are different things. Each plugin has clone function inside this is thread-safe (it's tested by https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/shared/include/behavior/core_threading_tests.hpp as well)

Reply:
Ok, so assuming that no plugin changes the original network and this is tested, and that the loading is tested with multi-threading, we can remove the cloning

Reply:
@ilyachur, @ilya-lavrenov What is the best way to forward `const ICNNNetwork&` network to `ICore::LoadNetwork` that get network as `const CNNNetwork&` ?

Reply:
Maybe via:
```cpp
explicit CNNNetwork(std::shared_ptr<ICNNNetwork> network)
```
with empty deleter for shared_ptr